src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -28,13 +28,8 @@ export default BaseAuthenticatorView.extend({
   initialize () {
     BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
 
-    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
-    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
-    const existingAuthenticator = _.find(authenticatorEnrollments, { id: currentAuthenticator.id });
-
-    // If the authenticator is already enrolled, we're in a challenge flow as opposed to enrollment
-    this.options.isChallenge = existingAuthenticator || false;
-    this.options.displayName = currentAuthenticator.displayName;
+    this.options.isChallenge = this.options.appState.isAuthenticatorChallenge();
+    this.options.displayName = this.options.appState.get('currentAuthenticator').displayName;
```

Comment:
`currentAuthenticator` v.s `currentAuthenticatorEnrollment`?

Reply:
This gets tricky based on if the enrolled authenticator is coming from `currentAuthenticatorEnrollment` or the list of `authenticatorEnrollments`. I think pulling this from the `currentAuthenticator` is our best bet.